### PR TITLE
add graphviz for network diagram

### PR DESCRIPTION
**Pull request by butlerx**
ive added the viz library in because i don't think we need to get in to bower or npm. if for some unseen future reason that npm is added viz can be moved to a package.json then

see #72

### CODE REVIEW

**Comment by butlerx** [comment]
https://md.redbrick.dcu.ie/s/SJ2JrfXM- example of diagrams rendered

File: docs/js/graphviz.js [hunk 1]
```diff
@@ -0,0 +1,16 @@
+var graphvizs = $('.section').find('code.graphviz')
+graphvizs.each(function (key, value) {
```
Comment (VoyTechnology) [inline docs/js/graphviz.js:2]: Can we use ES6? `(key, value) => {`

File: docs/js/graphviz.js [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+  try {
+    var $value = $(value)
```
Comment (VoyTechnology) [inline docs/js/graphviz.js:4]: s/`var`/`let`

File: docs/js/graphviz.js [hunk 3]
```diff
@@ -0,0 +1,16 @@
+var graphvizs = $('.section').find('code.graphviz')
```
Comment (VoyTechnology) [inline docs/js/graphviz.js:1]: Where is jQuery added?

**Review by VoyTechnology — DISMISSED** [review]
Is is somehow possible for us to include grapviz while not adding the actual actual library? Can we use a hosted version? Git submodules?

File: docs/js/graphviz.js (hunk 1 above)
Comment (butlerx) [reply docs/js/graphviz.js:2]: not all browsers support es6

File: docs/js/graphviz.js (hunk 3 above)
Comment (butlerx) [reply docs/js/graphviz.js:1]: already in theme

File: docs/js/graphviz.js (hunk 2 above)
Comment (butlerx) [reply docs/js/graphviz.js:4]: again not all browsers support let

**Review by butlerx — COMMENTED** [review]
this is front end code so needs to be backwards compatible theres no babel

**Comment by butlerx** [comment]
we cant  use submodules as readthedocs wont pull it in and halfpints build process isn't set up for it

File: docs/js/graphviz.js (hunk 1 above)
Comment (VoyTechnology) [reply docs/js/graphviz.js:2]: Looking at compatibility graph the only browser that doesn't support is is IE 11 and below

File: docs/js/graphviz.js (hunk 2 above)
Comment (VoyTechnology) [reply docs/js/graphviz.js:4]: IE 11 supports it

File: docs/js/graphviz.js (hunk 1 above)
Comment (butlerx) [reply docs/js/graphviz.js:2]: and portions of mobile browser
id rather have compatibility  then the ability to say its es6

**Comment by zergless** [comment]
I think adding the 200~ lines to the repo is worth. 

Those graphs looks v nice

**Review by CSmartt — APPROVED** [review]
Looks good to me. +1 for keeping browser compatibility as broad as possible also, bearing in mind docs are going to be read on mobile pretty often.

**Pull request merged**